### PR TITLE
Update actor prose

### PR DESCRIPTION
**Pull request by zixuanzh**
Update readme to reflect latest actor code changes per #774, notable changes:

- [x] simplify sector state (PreCommit, Active, and TemporaryFault)
- [x] simplify fault and recovery (TemporaryFault for a particular sector and DetectedFault for the entire miner)
- [x] update sector state and miner post state diagram
- [x] lazy deal payment and slashing evaluation
- [x] cron checks on different expiry events (PreCommit, Sector, SurpriseChallenge)
- [x] power and reward from a sector are potentially determined by sector duration, sector size and deal weight (they may not use all inputs)
- [x] there are two types of sectors (regular sectors with deals and CommittedCapacity sectors with no deals)
- [x] all sectors require explicit Expiration Epoch declared on PreCommit and Start epoch of a sector is determined by the ProveCommit epoch it gets added to the chain
- [x] miners now gain power from a sector immediately after ProveCommit
- [x] power of a sector remains constant throughout its lifetime
- [x] miners can extend the lifetime of a sector

### CODE REVIEW

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 1]
```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
 
 {{< readfile file="/docs/actors/builtin/storage_power/storage_power_actor_code.go" code="true" lang="go" >}}
 
+{{<label storage_power>}}
+## On Power
 
-{{<label power_table>}}
-# The Power Table
+Claimed power is assigned to every sector as a static function of its `SectorStorageWeightDesc` which includes `SectorSize`, `Duration`, and `DealWeight`. DealWeight is a measure that maps size and duration of active deals in a sector during its lifetime to its impact on power and reward distribution. A CommittedCapacity Sector will have a DealWeight of zero but all sectors have an explicit Duration which is defined from the ChainEpoch that the sector comes online in a ProveCommit message to the Expiration ChainEpoch of the sector. In principle, power is the number of votes a miner has in leader election and it is a point in time concept of storage. However, the exact function that maps `SectorStorageWeightDesc` to claimed `StoragePower` and `BlockReward` will be announced soon.
```
Comment (sternhenri) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: Link to `committedCapacity` definition

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 2]
```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
 
 {{< readfile file="/docs/actors/builtin/storage_power/storage_power_actor_code.go" code="true" lang="go" >}}
 
+{{<label storage_power>}}
+## On Power
```
Comment (sternhenri) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: why the move to here? I feel like it is more general than just the actor...

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 3]
```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
...
 
-The portion of blocks a given miner generates through leader election in EC (and so the block rewards they earn) is proportional to their `Power Fraction` over time. That is, a miner whose storage represents 1% of total storage on the network should mine 1% of blocks on expectation.
+Nominal power is what goes into consensus from claimed power. This includes checks on whether miner's claimed power has met minimum miner size requirement for consensus, whether miner is in DetectedFault state which will result in loss of all power, and whether miner is under collateralized.
```
Comment (sternhenri) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: General note that this nominal power vs claimed power vs others seems overly complicated to me...

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 4]
```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
...
-    - Power is decremented immediately after faults are declared, proportional to the faulty sector size.
-    - Power is incremented after a PoSt recovering from a fault.
-    - Power is definitively removed from the Power Table past the sector failure timeout (see {{<sref storage_faults>}})
+    - Power is incremented at SectorProveCommit
+    - All Power of a particular miner is decremented immediately after a missed SurprisePoSt (DetectedFault).
+    - A particular sector's power is decremented `OnSectorTemporaryFaultEffectiveBegin` when TemporaryFault is declared on the sector.
+    - A particular sector's power is added back `OnSectorTemporaryFaultEffectiveEnd` and miner is expected to prove over this sector. 
+    - A particular sector's power is removed when the sector is terminated through sector expiration or miner invocation (`_rtTerminateSector`).
 To summarize, only sectors in the Active state will command power. A Sector becomes Active after their first PoSt from Committed and Recovering stages. Power is immediately decremented when an Active Sector enters the Failing state (through DeclareFaults or Cron) and when an Active Sector expires.
```
Comment (sternhenri) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: [blocking] Note that with ePoSt this is no longer necessary. Sectors can become active right after the commitment. No need to wait for the first PoSt anymore (since you can only profit from an active sector through ePoSt)

File: src/systems/filecoin_markets/storage_market/storage_deal/storage_deal_flow.md [hunk 5]
```diff
@@ -8,48 +8,41 @@ title: Storage Deal Flow
...
-- 11. For all sectors that are off from the `ProvingSet`, these sectors are failing. Increment `FaultCount` on these sectors and if any of these sectors are failing for `MAX_CONSECUTIVE_FAULTS` consecutive `ChainEpoch`, these sectors are terminated and cleared from the network.
-- 13. Process sector expiration. Sectors expire when all deals in that sector have expired. Expired sectors will be cleared and `StorageDealCollateral` for both miners and users returned depending on the state that the sectors are in.
-- 14. Submit `FaultReport` and `PowerReport` to `StoragePowerActor` for slashing and power accounting.
-- 15. Check and ensure that Pledge Collateral is statisfied. TODO: some details are missing here, also related to ProvingPeriod depending on PoSt construction.
-- 16. Update challenge status and add `Committed` sectors received during the challenge to the `Sectors`, `ProvingSet`, and `SectorTable`.
-- 17. All Sectors will be considered in `DetectedFaults` when a miner fail to `SubmitSurprisePoSt` or `SubmitElectionPoSt` (jointly `SubmitPoSt` methods) in a proving period and detected by `onMissedSurprisePoSt` in `CheckSurprisePoStSubmissionHappened` (move to State 18).
```
Comment (sternhenri) [inline src/systems/filecoin_markets/storage_market/storage_deal/storage_deal_flow.md]: Are these steps no longer needed?

File: src/systems/filecoin_mining/storage_mining/_index.md [hunk 6]
```diff
@@ -19,36 +19,44 @@ The Filecoin Storage Mining Subsystem ensures a storage miner can effectively co
 The above involves a number of steps to putting on and maintaining online storage, such as:
 
 - Committing new storage (see Sealing and PoRep)
-- Continously proving storage ({see {<sref election_post>}})
+- Continously proving storage
```
Comment (sternhenri) [inline src/systems/filecoin_mining/storage_mining/_index.md]: why the link removal?
```suggestion
- Continously proving storage (see {{<sref election_post>}})
```

File: src/systems/filecoin_mining/storage_mining/_index.md [hunk 7]
```diff
@@ -19,36 +19,44 @@ The Filecoin Storage Mining Subsystem ensures a storage miner can effectively co
...
 - Declaring storage faults and recovering from them.
 
-## Storage States
+## Sector Types
+
+There are two types of sectors, Regular Sectors with storage deals in them and Committed Capacity (CC) Sectors with no deals. All sectors require an expiration epoch that is declared upon PreCommit and sectors are assigned a StartEpoch at ProveCommit. Start and Expiration epoch collectively define the lifetime of a Sector. Length and size of active deals in a sector's lifetime determine the `DealWeight` of the sector. `SectorSize`, `Duration`, and `DealWeight` statically determine the power assigned to a sector that will remain constant throughout its lifetime. More details on cost and reward for different sector types will be announced soon.
+
```
Comment (sternhenri) [inline src/systems/filecoin_mining/storage_mining/_index.md]: Explain why you would set up one sector type vs the other?

File: src/systems/filecoin_mining/storage_mining/_index.md [hunk 8]
```diff
@@ -19,36 +19,44 @@ The Filecoin Storage Mining Subsystem ensures a storage miner can effectively co
...
 
-Through most of this cycle, sectors will be part of the miner's `Proving Set`. This is the set of sectors that the miners are supposed to generate proofs against. It includes:
-- `Committed Sectors` which have an associated PoRep but have but yet to generate a PoSt, 
-- `Active Sectors` which have successfully been proven and are used in {{<sref storage_power_consensus>}} (SPC), and
-- `Recovering Sectors` which were faulted and are poised to recover through a new PoSt.
+There are three states that an individual sector can be in:
+
+- `PreCommit` when a sector has been added through a PreCommit message.
+- `Active` when a sector has been proven through a ProveCommit message and when a sector's TemporaryFault period has ended.
+- `TemporaryFault` when a miner declares fault on a particular sector.
```
Comment (sternhenri) [inline src/systems/filecoin_mining/storage_mining/_index.md]: Great!

File: src/systems/filecoin_mining/storage_mining/_index.md [hunk 9]
```diff
@@ -19,36 +19,44 @@ The Filecoin Storage Mining Subsystem ensures a storage miner can effectively co
...
-We illustrate these states below.
+### Miner PoSt State
 
-{{< readfile file="storage_mining_subsystem.id" code="true" lang="go" >}}
+`MinerPoStState` keeps track of a miner's state in responding to PoSt and there are three states in `MinerPoStState`:
 
-### Sector in StorageMiner State Machine (new one)
+- `OK` miner has passed either a ElectionPoSt or a SurprisePoSt sufficiently recently.
+- `Challenged` miner has been selected to prove its storage via SurprisePoSt and is currently in the Challenged state
+- `DetectedFault` miner has failed at least one SurprisePoSt, indicating that all claimed storage may not be proven. Miner has lost power on its sector and recovery can only proceed by a successful response to a subsequent SurprisePoSt challenge, up until the limit of number of consecutive failures.
```
Comment (sternhenri) [inline src/systems/filecoin_mining/storage_mining/_index.md]: This is very helpful

Comment (sternhenri) [reply src/systems/filecoin_mining/storage_mining/_index.md]: [blocking] Aren't we missing one, ie `DeclaredFault`?

File: src/systems/filecoin_mining/storage_mining/_index.md [hunk 10]
```diff
@@ -19,36 +19,44 @@ The Filecoin Storage Mining Subsystem ensures a storage miner can effectively co
...
 
-{{< diagram src="diagrams/sector_state_fsm.dot.svg" title="Sector State (new one)" >}}
+`DetectedFault` is a miner-wide PoSt state when all sectors are considered inactive. All power is lost immediately and pledge collateral is slashed. If a miner remains in `DetectedFault` for more than MaxConsecutiveFailures, all sectors will be terminated, both power and market actors will be notified.
 
-{{< diagram src="diagrams/sector_state_legend.dot.svg" title="Sector State Legend (new one)" >}}
+`ProvingSet` is consist of sectors that miners are required to generate proofs against and is what counts towards miners' power. In other words, `ProvingSet` is a set of all `Active` sectors for a particular miner. `ProvingSet` is only relevant when the miner is in OK stage of its `MinerPoStState`. When a miner is in the `Challenged` state, `ChallengedSectors` specify the list of sectors to be challenged which is the `ProvingSet` before the challenge is issued thus allowing more sectors to be added while it is in the `Challenged` state.
```
Comment (sternhenri) [inline src/systems/filecoin_mining/storage_mining/_index.md]: - typo s/is consist/consists
This sentence: 
> When a miner is in the `Challenged` state, ...

is a bit confusing. would we want to allow miners to add more sectors while in the challenged state? And what of detected vs declared faults?

File: src/systems/filecoin_mining/storage_mining/mining_cycle.md [hunk 11]
```diff
@@ -68,23 +68,15 @@ The miner actor has two distinct 'controller' {{<sref app_address "addresses">}}
...
+Note that any change to worker keys after registration must be appropriately delayed in relation to randomness lookback for SEALing data (see [this issue](https://github.com/filecoin-project/specs/issues/415)).
 
 ### Step 1: Committing Sectors
 
-When the miner has completed their first seal, they should post it on-chain using the {{<sref storage_miner_actor>}}'s `ProveCommitSector` function. The miner will need to put up pledge collateral in proportion to the amount of storage they commit on chain. If the miner had zero committed sectors prior to this call, this begins their proving period.
+When the miner has completed their first seal, they should post it on-chain using the {{<sref storage_miner_actor>}}'s `ProveCommitSector` function. The miner will need to put up pledge collateral in proportion to the amount of storage they commit on chain. If the miner had zero committed sectors prior to this call, this begins their proving period and miner gains power for the sector.
 
 You can read more about sectors {{<sref sector "here">}}.
 
-### Step 2: Getting Power
```
Comment (sternhenri) [inline src/systems/filecoin_mining/storage_mining/mining_cycle.md]: even if this is removed, there needs to be something tying proving sectors to gaining power...

File: src/systems/filecoin_mining/storage_mining/mining_cycle.md [hunk 12]
```diff
@@ -100,9 +92,9 @@ In this period, the miner can still:
...
+If a miner detects {{<sref storage_faults>}} among their sectors (any sort of storage failure that would prevent them from crafting a PoSt), they should declare these faults with the `DeclareTemporaryFaults()` method of the Storage Miner Actor. 
 
-The miner will be unable to craft valid PoSts over faulty sectors, thereby reducing their chances of winning Election and SurprisePoSts. By declaring a fault, the miner will no longer be challenged on that sector, and will lose power accordingly. The miner can reverse the fault by calling `RecoverFaults()` to add the sector back into the `ProvingSet` and regain power.
+The miner will be unable to craft valid PoSts over faulty sectors, thereby reducing their chances of winning Election and SurprisePoSts. By declaring a fault, the miner will no longer be challenged on that sector, and will lose power accordingly. The miner can specify how long the duration of their TemporaryFault and pay a TemporaryFaultFee.
```
Comment (sternhenri) [inline src/systems/filecoin_mining/storage_mining/mining_cycle.md]: [blocking] I am not sure specifying how long the duration of the fault makes sense. These should be unplanned storage outages, and duration should not be predictable. Accordingly, we should just set the fee to the max amount to be conservative, but treat all unplanned outages the same for simplicity.

File: src/systems/filecoin_vm/sysactors/reward_actor.md [hunk 13]
```diff
@@ -2,9 +2,7 @@
...
+RewardActor is where unminted Filecoin tokens are kept. RewardActor contains a `RewardMap` which is a mapping from owner addresses to `Reward` structs. `Reward` struct is created to preserve the flexibility of introducing block reward vesting into the protocol. `MintReward` creates a new `Reward` struct and adds it to the `RewardMap`. A `Reward` struct contains a `StartEpoch` that keeps track of when this `Reward` is created, `Value` that represents the total number of tokens rewarded, and `EndEpoch` which is when the reward will be fully vested. `VestingFunction` is currently an enum to represent the flexibility of different vesting functions. `AmountWithdrawn` records how many tokens have been withdrawn from a `Reward` struct so far. Owner addresses can call `WithdrawReward` which will withdraw all vested tokens that the investor address has from the RewardMap so far. When `AmountWithdrawn` equals `Value` in a `Reward` struct, the `Reward` struct will be removed from the `RewardMap`.
```
Comment (sternhenri) [inline src/systems/filecoin_vm/sysactors/reward_actor.md]: Add spaces to make this less of a huge block of text.

File: src/systems/filecoin_markets/storage_market/storage_deal/storage_deal_flow.md (hunk 5 above)
Comment (zixuanzh) [reply src/systems/filecoin_markets/storage_market/storage_deal/storage_deal_flow.md]: They have been covered in other steps. Deal payment, sector expiration and slashing are all done lazily.

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 14]
```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
...
 
-The portion of blocks a given miner generates through leader election in EC (and so the block rewards they earn) is proportional to their `Power Fraction` over time. That is, a miner whose storage represents 1% of total storage on the network should mine 1% of blocks on expectation.
+Nominal power is what goes into consensus from claimed power. This includes checks on whether miner's claimed power has met minimum miner size requirement for consensus, whether miner is in DetectedFault state which will result in loss of all power, and whether miner is under collateralized.
+
```
Comment (jzimmerman) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: > Nominal power is what goes into consensus from claimed power. This includes checks on whether miner's claimed power has met minimum miner size requirement for consensus, whether miner is in DetectedFault state which will result in loss of all power, and whether miner is under collateralized.

I think this is not quite right -- some of the checks don't affect nominal power. More precisely:

- Nominal power = claimed power, unless the miner is in DetectedFault state. Nominal power is used to determine total network storage power for purposes of consensus minimum.
- Consensus power = nominal power, unless the miner fails to meet consensus minimum, or is undercollateralized.

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 15]
```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
...
+{{<label min_miner_size>}}
+## Minimum Miner Size
+
+In order to secure Storage Power Consensus, the system defines a minimum miner size required to participate in consensus.
+
+Specifically, miners must have either at least `MIN_MINER_SIZE_STOR` of power (i.e. storage power currently used in storage deals) in order to participate in leader election. If no miner has `MIN_MINER_SIZE_STOR` or more power, miners with at least as much power as the smallest miner in the top `MIN_MINER_SIZE_TARG` of miners (sorted by storage power) will be able to participate in leader election. In plain english, take `MIN_MINER_SIZE_TARG = 3` for instance, this means that miners with at least as much power as the 3rd largest miner will be eligible to participate in consensus.
+
+Miners smaller than this cannot mine blocks and earn block rewards in the network. Their power will not be counted as part of total network power. However, **it is important to note that such miners can still have their power faulted and be penalized accordingly**.
```
Comment (jzimmerman) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: As above -- I believe, as we've currently implemented it, their power will still be counted in the total network (raw) storage power, even though it won't yield votes in leader election.

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 16]
```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
...
 
-SPC provides a power table abstraction which tracks miner power (i.e. miner storage in relation to network storage) over time. The power table is updated for new sector commitments (incrementing miner power), when PoSts fail to be put on-chain (decrementing miner power) or for other storage and consensus faults.
+Accordingly, to bootstrap the network, the genesis block must include miners taking part in valid storage deals along with appropriate committed storage.
```
Comment (jzimmerman) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: I don't think deals are necessary here -- the genesis block might just contain CC sectors.

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 17]
```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
...
 
-Querying the power table for this miner at different rounds should yield (using the following shorthand as an illustration only):
-- `Power(X, 90) == 0`
-- `Power(X, 100) == 0`
-- `Power(X, 110) == 0`
-- `Power(X, 111) == 10`
-- `Power(X, 120) == 10`
-- `Power(X, 135) == 10`
-- `Power(x, 136) == 20`
+- `MIN_MINER_SIZE_STOR = 1 << 40 Bytes` (100 TiB)
```
Comment (jzimmerman) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: minor: `100 * (1 << 40)`

File: src/systems/filecoin_mining/storage_mining/mining_cycle.md [hunk 18]
```diff
@@ -68,23 +68,15 @@ The miner actor has two distinct 'controller' {{<sref app_address "addresses">}}
...
+Note that any change to worker keys after registration must be appropriately delayed in relation to randomness lookback for SEALing data (see [this issue](https://github.com/filecoin-project/specs/issues/415)).
 
 ### Step 1: Committing Sectors
 
-When the miner has completed their first seal, they should post it on-chain using the {{<sref storage_miner_actor>}}'s `ProveCommitSector` function. The miner will need to put up pledge collateral in proportion to the amount of storage they commit on chain. If the miner had zero committed sectors prior to this call, this begins their proving period.
+When the miner has completed their first seal, they should post it on-chain using the {{<sref storage_miner_actor>}}'s `ProveCommitSector` function. The miner will need to put up pledge collateral in proportion to the amount of storage they commit on chain. If the miner had zero committed sectors prior to this call, this begins their proving period and miner gains power for the sector.
```
Comment (jzimmerman) [inline src/systems/filecoin_mining/storage_mining/mining_cycle.md]: minor: `proving period` no longer applies with ElectionPoSt

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md (hunk 2 above)
Comment (zixuanzh) [reply src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: moved there because thats where power was introduced but when I was linking the docs, think it makes sense to exist at where it was. It has been moved back.

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md (hunk 3 above)
Comment (zixuanzh) [reply src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: this has been updated per Joe's comment. Hope it clarifies.

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md (hunk 4 above)
Comment (zixuanzh) [reply src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: Ack and it has been updated.

File: src/systems/filecoin_mining/storage_mining/_index.md (hunk 6 above)
Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/_index.md]: link was inactive and it has been restored.

File: src/systems/filecoin_mining/storage_mining/_index.md (hunk 7 above)
Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/_index.md]: it depends on the cost and reward structure for both sectors which will be announced soon.

File: src/systems/filecoin_mining/storage_mining/_index.md (hunk 9 above)
Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/_index.md]: DeclaredFault (DeclareTemporaryFault) is per individual sector and DetectedFault is applied to all sectors of a particular miner and hence on MinerPoStState.

File: src/systems/filecoin_mining/storage_mining/_index.md (hunk 10 above)
Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/_index.md]: typo has been fixed.

per second comment, we will still want to allow miners to add more sectors while in challenged state (this is a requirement from impl). I think this wont affect Challenged state since it specifies a list of ChallengedSectors. Miners should not be allowed to declare fault when it is in DetectedFault state. I have flagged this to @jzimmerman and think this deserves a bit more discussion. We can come back to this in a separate PR.

File: src/systems/filecoin_mining/storage_mining/mining_cycle.md (hunk 11 above)
Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/mining_cycle.md]: added link to the section where power is discussed.

File: src/systems/filecoin_mining/storage_mining/mining_cycle.md (hunk 12 above)
Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/mining_cycle.md]: This feels like a policy discussion and agree that it can potentially be problematic when the fault duration is too long. Lets discuss this in sync? cc @jzimmerman

File: src/systems/filecoin_vm/sysactors/reward_actor.md (hunk 13 above)
Comment (zixuanzh) [reply src/systems/filecoin_vm/sysactors/reward_actor.md]: done, thanks

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md (hunk 15 above)
Comment (zixuanzh) [reply src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: thanks, it has been updated.

**Comment by mishmosh** [comment]
Nice prose additions! For PR name, do you mean "update markdown" vs. "update readme"?

File: src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md [hunk 19]
```diff
@@ -14,58 +14,37 @@ title: Storage Power Actor
...
-
-Put another way, power accounting in the SPC is delayed between storage being proven or faulted, and power being updated in the power table (and so for leader election). This ensures fairness over time.
+Sector ProveCommit is the first time power is proven to the network and hence power is first added through `_rtAddPowerForSector` at `OnSectorProveCommit`. Power is also added when a sector's TemporaryFault period has ended. Miners are expected to prove over all their sectors that contribute to their power. `_rtDeductClaimedPowerForSectorAssert` is called to decrement a miner's power. This is called when a sector expires or invoked by miner through `OnSectorTerminate` and when a sector enters TemporaryFault through `OnSectorTemporaryFaultEffectiveBegin`. Both `_rtAddPowerForSector` and `_rtDeductClaimedPowerForSectorAssert` are currently called at `OnSectorModifyWeightDesc` as power is determined by `SectorStorageWeightDesc` and `SectorStorageWeightDesc` is only modified when a miner calls `ExtendSectorExpiration` to extend a sector's duration. This may or may not have an impact on power but the machinery is in place to preserve the flexibility.
```
Comment (anorth) [inline src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md]: Nit: referencing internal method names here is extremely likely to rot. Better to keep the description conceptual.

File: src/systems/filecoin_mining/storage_mining/_index.md [hunk 20]
```diff
@@ -19,36 +18,44 @@ The Filecoin Storage Mining Subsystem ensures a storage miner can effectively co
...
-{{< readfile file="storage_mining_subsystem.id" code="true" lang="go" >}}
+`MinerPoStState` keeps track of a miner's state in responding to PoSt and there are three states in `MinerPoStState`:
 
-### Sector in StorageMiner State Machine (new one)
+- `OK` miner has passed either a ElectionPoSt or a SurprisePoSt sufficiently recently.
+- `Challenged` miner has been selected to prove its storage via SurprisePoSt and is currently in the Challenged state
+- `DetectedFault` miner has failed at least one SurprisePoSt, indicating that all claimed storage may not be proven. Miner has lost power on its sector and recovery can only proceed by a successful response to a subsequent SurprisePoSt challenge, up until the limit of number of consecutive failures.
 
-{{< diagram src="diagrams/sector_state_fsm.dot.svg" title="Sector State (new one)" >}}
+`DetectedFault` is a miner-wide PoSt state when all sectors are considered inactive. All power is lost immediately and pledge collateral is slashed. If a miner remains in `DetectedFault` for more than MaxConsecutiveFailures, all sectors will be terminated, both power and market actors will be notified.
```
Comment (anorth) [inline src/systems/filecoin_mining/storage_mining/_index.md]: If power was lost when challenged (not documented, but I heard it verbally), it's not lost (again) when the challenge expires un-met.

File: src/systems/filecoin_blockchain/storage_power_consensus/_index.md [hunk 21]
```diff
@@ -33,6 +33,16 @@ We must distinguish between both types of "miners" (storage and block miners). {
...
+{{<label storage_power>}}
+## On Power
+
+Claimed power is assigned to every sector as a static function of its `SectorStorageWeightDesc` which includes `SectorSize`, `Duration`, and `DealWeight`. DealWeight is a measure that maps size and duration of active deals in a sector during its lifetime to its impact on power and reward distribution. A CommittedCapacity Sector (see {{<sref storage_mining_subsystem "Sector Types">}}) will have a DealWeight of zero but all sectors have an explicit Duration which is defined from the ChainEpoch that the sector comes online in a ProveCommit message to the Expiration ChainEpoch of the sector. In principle, power is the number of votes a miner has in leader election and it is a point in time concept of storage. However, the exact function that maps `SectorStorageWeightDesc` to claimed `StoragePower` and `BlockReward` will be announced soon.
+
+More precisely,
+
+- Nominal power = claimed power, unless the miner is in DetectedFault state. Nominal power is used to determine total network storage power for purposes of consensus minimum.
+- Consensus power = nominal power, unless the miner fails to meet consensus minimum, or is undercollateralized.
```
Comment (sternhenri) [inline src/systems/filecoin_blockchain/storage_power_consensus/_index.md]: This is great

Comment (sternhenri) [reply src/systems/filecoin_blockchain/storage_power_consensus/_index.md]: nit: could add `claimed power (to add how declared faults factor in)` above but looks great.

File: src/systems/filecoin_mining/storage_mining/_index.md [hunk 22]
```diff
@@ -54,7 +53,7 @@ A particular sector enters `TemporaryFault` from `Active` through `DeclareTempor
...
 
-`ProvingSet` is consist of sectors that miners are required to generate proofs against and is what counts towards miners' power. In other words, `ProvingSet` is a set of all `Active` sectors for a particular miner. `ProvingSet` is only relevant when the miner is in OK stage of its `MinerPoStState`. When a miner is in the `Challenged` state, `ChallengedSectors` specify the list of sectors to be challenged which is the `ProvingSet` before the challenge is issued thus allowing more sectors to be added while it is in the `Challenged` state.
+`ProvingSet` consists of sectors that miners are required to generate proofs against and is what counts towards miners' power. In other words, `ProvingSet` is a set of all `Active` sectors for a particular miner. `ProvingSet` is only relevant when the miner is in OK stage of its `MinerPoStState`. When a miner is in the `Challenged` state, `ChallengedSectors` specify the list of sectors to be challenged which is the `ProvingSet` before the challenge is issued thus allowing more sectors to be added while it is in the `Challenged` state.
```
Comment (sternhenri) [inline src/systems/filecoin_mining/storage_mining/_index.md]: maybe worth explicitly saying but
> sectors cannot be removed from `ProvingSet` when `Challenged`

File: src/systems/filecoin_mining/storage_mining/_index.md (hunk 10 above)
Comment (sternhenri) [reply src/systems/filecoin_mining/storage_mining/_index.md]: I agree with all your statements here. I think in order to ensure faults are declared appropriately (and to incentivize it) we should default to what you said (ie cant declare fault if already detected). I would add now (+ in code) but understand deferral.

File: src/systems/filecoin_mining/storage_mining/mining_cycle.md (hunk 12 above)
Comment (sternhenri) [reply src/systems/filecoin_mining/storage_mining/mining_cycle.md]: Yup. stand by it but let's defer

File: src/systems/filecoin_markets/storage_market/storage_deal/storage_deal_flow.md [hunk 23]
```diff
@@ -8,48 +8,42 @@ title: Storage Deal Flow
...
-- 6. Miners are not allowed to call `DeclareFaults` or `RecoverFaults` when they are in the `Challneged` state but `CommitSector` is allowed and sectors will be added to a `StagedCommittedSectors` buffer. When miners get out of the `Challenged` status, `StagedCommittedSectors` will be copied over to their `Sectors`, `ProvingSet` and `SectorTable` and emptied.
+- 5. Miners enter the `Challenged` status when it receives a SurprisePoSt challenge from the chain. Miners will then have X Epoch as the ProvingPeriod to submit a successful PoSt before the chain checks for SurprisePoSt expiry. Miners can only get out the challenge with `SubmitSurprisePoStResponse`.
+- 6. Miners are allowed to DeclareTemporaryFault when they are in the `Challenged` state but this will not change the list of sectors challenged as `Challenged` state specifies a list of sectors to be challenged which is a snapshot of all Active sectors at the time of challenge. Miners are also allowed to call ProveCommit which will add to their ClaimedPower but their Nominal and Consensus Power are still zero whe  they are in either Challenged or DetectedFault state.
```
Comment (sternhenri) [inline src/systems/filecoin_markets/storage_market/storage_deal/storage_deal_flow.md]: Doesn't make sense to allow the declaration to me...

Typo
> Consensus Power are still zero when they are

Comment (zixuanzh) [reply src/systems/filecoin_markets/storage_market/storage_deal/storage_deal_flow.md]: thanks, Nominal and Consensus*

File: src/systems/filecoin_mining/storage_mining/_index.md (hunk 22 above)
Comment (zixuanzh) [reply src/systems/filecoin_mining/storage_mining/_index.md]: the list of Challenged sectors specified in Challenged state is copy of the ProvingSet at the time of challenge so hence it can't be added to or removed from.

Comment (sternhenri) [reply src/systems/filecoin_mining/storage_mining/_index.md]: Then the above sentence is slightly confusing, esp
> allowing more sectors to be added to it...

File: src/systems/filecoin_blockchain/storage_power_consensus/_index.md (hunk 21 above)
Comment (zixuanzh) [reply src/systems/filecoin_blockchain/storage_power_consensus/_index.md]: thanks and it has been updated.
